multidict/_multidict.c

```diff
@@ -166,6 +178,20 @@ _multidict_append_items_seq(MultiDictObject *self, PyObject *arg,
     }
 
     return 0;
+invalid_type:
+    PyErr_Format(
+        PyExc_TypeError,
+        "%s takes either dict or list of (key, value) pairs",
+        name,
+        NULL
+    );
+    goto fail;
```

Comment:
This is not needed

Reply:
Indeed, you are right.
I believe this goto increases readability a little, the C compiler optimizer will drop this instruction anyway.